tests/components/smart_meter_b_route/__init__.py

```diff
@@ -0,0 +1,27 @@
+"""Tests for the Smart Meter B-route integration."""
+
+from homeassistant.components.smart_meter_b_route.const import DOMAIN
+from homeassistant.components.smart_meter_b_route.coordinator import (
+    BRouteUpdateCoordinator,
+)
+from homeassistant.const import CONF_DEVICE, CONF_ID, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+user_input = {
+    CONF_DEVICE: "/dev/ttyUSB42",
+    CONF_ID: "B_ROUTE_ID",
+    CONF_PASSWORD: "B_ROUTE_PASSWORD",
+}
+
+
+def configure_integration(hass: HomeAssistant) -> MockConfigEntry:
+    """Configure the integration."""
+    entry = MockConfigEntry(
+        domain=DOMAIN, data=user_input, entry_id="123456", unique_id="123456"
+    )
+    entry.runtime_data = BRouteUpdateCoordinator(hass, "device", "id", "password")
+    entry.add_to_hass(hass)
+
+    return entry
```

Comment:
Instead of assigning the coordinator to the entry like this, we should mock out the library and setup HA like normal. The integration itself should take care of creating a coordinator and assigning it. Our test look more like end-to-end tests, but they make sure the use cases are still working

Reply:
@joostlek Sorry, do you have an example of a test with this flow ? I am not sure how I should proceed... I do not know what is normal yet lol